nni/compression/pytorch/pruning/basic_pruner.py

```diff
@@ -755,11 +755,20 @@ def collect_activation(_module: Module, _input: Tensor, output: Tensor):
                 _module = _module.module
             batch_dims, batch_num = get_output_batch_dims(output, _module)  # type: ignore
             activation = self._activation_trans(output, batch_dims)
+            batch_num = torch.tensor([batch_num]).to(activation.device)
+            if self.is_ddp_model:
+                import torch.distributed as dist
+                batch_num = batch_num.clone()
+                activation = activation.clone()
```

Comment:
do we need detach() here?

Reply:
I think there is no need to use .detach(), the operation here is to average the grads on multiple GPUs, without changing the graph structure of the variables.

Reply:
if we don't not detach `activation`, will it be recorded as a data that for backward needs and never released?